### ISSUE TITLE
feat: update ES mapping with filter fields

### ISSUE THREAD

**Issue by jupyterjazz** [jina-ai/now#913]
Filtering in ES doesn't work as expected because we don't define filter fields in the Mapping. 
The reason is that they automatically get assigned a `type text` and are then analyzed in some ways. 
As a result we currently have this strange behavior:

Indexed: Piet Mondrian        Search by Piet Mondrian           FAILED
Indexed: Piet Mondrian        Search by piet mondrian           FAILED
Indexed: Piet Mondrian        Search by piet                            SUCCESS
Indexed: Piet Mondrian        Search by mondrian                   SUCCESS
Indexed: Piet Mondrian        Search by mondri                       SUCCESS

So basically is successfully filters on subwords but not really.

I tested and the solution is to define the filter fields in the mappings as `type keyword`

UAC:

- [ ] filter fields (or even just non-index fields) are collected in the user input
- [ ] user input is passed to the indexer (Joschka's PR)
- [ ] updating the mapping to include filter fields too
